packages/core/core/src/Transformation.js

```diff
@@ -140,13 +140,13 @@ export default class Transformation {
           },
         ]);
       }
+    }
 
-      if (existing == null) {
-        // If no existing sourcemap was found, initialize asset.sourceContent
-        // with the original contents. This will be used when the transformer
-        // calls setMap to ensure the source content is in the sourcemap.
-        asset.sourceContent = await asset.getCode();
-      }
+    if (existing == null) {
+      // If no existing sourcemap was found, initialize asset.sourceContent
+      // with the original contents. This will be used when the transformer
+      // calls setMap to ensure the source content is in the sourcemap.
+      asset.sourceContent = await asset.getCode();
```

Comment:
Will this end up creating a string representation of binary content? Maybe we can do this on demand when calling setMap?

Reply:
This would be an issue as a common pattern is `let map = asset.getMap(); myMap.extend(map); asset.setMap(map);`, but by the time we lazily initialize a map inside setMap, getMap would already be called and it returns empty then.

Reply:
Well we only want to do it for the first transformer in a pipeline

Reply:
Isn't that a separate problem from getMap/setMap?

Reply:
Not sure if I'm missing something here, but wouldn't `asset.getCode()` return transpiled code by the time we get to `set/getMap`?